### PR TITLE
#1442 when importing a simulation set project simulation settings

### LINKED ISSUE #1422: Ongoing project conversion
- [ ] Maybe update due to https://github.com/Open-Systems-Pharmacology/PK-Sim/issues/2849
- [ ] https://github.com/Open-Systems-Pharmacology/PK-Sim/issues/2946
- [ ] https://github.com/Open-Systems-Pharmacology/MoBi/issues/1110

### LINKED ISSUE #1442: When importing a simulation from PK-Sim, set the project-wide simulation settings
When opening one pkml from PK-Sim, set the simulation settings (output intervals, solver settings, output selections) from this simulation to project wide defaults.

### PR DESCRIPTION
**Pull request by benjaperez1983**
Fixes #1442

# Description

When importing a simulation from PK-Sim, set the project-wide simulation settings

## Type of change

Please mark relevant options with an `x` in the brackets.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires documentation changes (link at least one [user](https://github.com/Open-Systems-Pharmacology/docs) or [developer](https://github.com/Open-Systems-Pharmacology/developer-docs) documentation issue):
- [ ] Algorithm update - updates algorithm documentation/questions/answers etc.
- [ ] Other (please describe):

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Integration tests
- [ ] Unit tests
- [ ] Manual tests 
- [ ] No tests required

# Reviewer checklist

Mark everything that needs to be checked before merging the PR.

- [ ] Check if the code is well documented
- [ ] Check if the behavior is what is expected
- [ ] Check if the code is well tested
- [ ] Check if the code is readable and well formatted
- [ ] Additional checks (document below if any)
- [ ] Check if documentation update issue(s) are created if the option `This change requires a documentation update` above is selected

# Screenshots (if appropriate):

# Questions (if appropriate):

### CODE REVIEW

File: src/MoBi.Presentation/Tasks/ProjectTask.cs [hunk 1]
```diff
@@ -93,23 +97,31 @@ public void StartWithJournal(string journalFilePath)
...
          SimulationTransfer simulationTransfer = null;
-
          _heavyWorkManager.Start(() => simulationTransfer = LoadSimulationTransferDataFromFile(fileName));
-         if (simulationTransfer == null)
-            return;
 
-         _context.AddToHistory(addSimulationTransferToProject(simulationTransfer));
-         loadJournalIfNotLoadedAlready(project, simulationTransfer.JournalPath);
-         notifyProjectLoaded();
+         if (simulationTransfer != null)
```
Comment (rwmcintosh) [inline src/MoBi.Presentation/Tasks/ProjectTask.cs:109]: Can you reverse the condition and return early instead?

File: tests/MoBi.Tests/Presentation/ProjectTaskSpecs.cs [hunk 2]
```diff
@@ -110,6 +121,45 @@ public void should_update_favorites_in_projects()
       {
          _simulationTransfer.Favorites.Each(favorite => _project.Favorites.ShouldContain(favorite));
       }
+
+      [Observation]
+      public void should_update_default_simulation_settings_in_project()
```
Comment (rwmcintosh) [inline tests/MoBi.Tests/Presentation/ProjectTaskSpecs.cs:126]: I think the naming is off here.

File: tests/MoBi.Tests/Presentation/ProjectTaskSpecs.cs [hunk 3]
```diff
@@ -110,6 +121,45 @@ public void should_update_favorites_in_projects()
...
+      {
+         A.CallTo(() => _interactionTasksForSimulationSettings.UpdateDefaultSimulationSettingsInProject(_simulationTransfer.Simulation.Settings.OutputSchema, _simulationTransfer.Simulation.Settings.Solver)).MustNotHaveHappened();
+      }
+
+      [Observation]
+      public void should_update_default_output_selections_in_project()
```
Comment (rwmcintosh) [inline tests/MoBi.Tests/Presentation/ProjectTaskSpecs.cs:132]: naming is off here

**Review by rwmcintosh — CHANGES_REQUESTED** [review]
Should we also trigger this simulation settings update if the user loads a simulation to an empty project?

If there is already a simulation in the project, don't update the settings, if there isn't one, then the project settings should be updated?

File: tests/MoBi.Tests/Presentation/ProjectTaskSpecs.cs (hunk 3 above)
Comment (benjaperez1983) [reply tests/MoBi.Tests/Presentation/ProjectTaskSpecs.cs:132]: Sorry, don`t get it. How should this be named then?

Comment (rwmcintosh) [reply tests/MoBi.Tests/Presentation/ProjectTaskSpecs.cs:132]: The names indicate that the simulation settings should be updated, but the tests are checking that the settings are not updated.

Comment (benjaperez1983) [reply tests/MoBi.Tests/Presentation/ProjectTaskSpecs.cs]: oh I See... sorry I`m dumb!

**Review by rwmcintosh — APPROVED** [review]

**Review by Yuri05 — APPROVED** [review]

**Pull request merged**